### PR TITLE
feat: Improve Data Visualizer with fixed period "Weekly (Start Sunday)" labels[2.42-DHIS2-17955]

### PR DESCRIPTION
**Pull request by d-bernat**
For the short names of the weekly period dimension, we were displaying a less meaningful version, such as "2024SunW1." To improve clarity, we are now updating the short name format to "W1 2023-12-31 - 2024-01-06" when the DHIS2 instance is set to display short names.

### CODE REVIEW

File: dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/generator/scenarios/aggregated.json [hunk 1]
```diff
@@ -48,6 +48,20 @@
       "version": {
         "min": 42
       }
+    },
+    {
+      "name": "weeklyShortNameLabel",
+      "query": "/api/42/analytics?dimension=dx%3AfbfJHSPpUQD,pe%3A2021SunW52&filter=ou%3AUSER_ORGUNIT&displayProperty=SHORTNAME&includeNumDen=true&skipMeta=false&skipData=false",
```
Comment (maikelarabori) [inline dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/generator/scenarios/aggregated.json:54]: Please, use decoded URLs

File: dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/generator/scenarios/aggregated.json [hunk 2]
```diff
@@ -48,6 +48,20 @@
...
+      "version": {
+        "min": 42
+      }
+    },
+    {
+      "name": "weeklyNameLabel",
+      "query": "/api/42/analytics?dimension=dx%3AfbfJHSPpUQD,pe%3A2021SunW52&filter=ou%3AUSER_ORGUNIT&displayProperty=NAME&includeNumDen=true&skipMeta=false&skipData=false",
```
Comment (maikelarabori) [inline dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/generator/scenarios/aggregated.json:61]: Please, use decoded URLs

**Review by maikelarabori — APPROVED** [review]
Looks, good. Just check the comment about the URLs.
Thx!

File: dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/generator/scenarios/aggregated.json (hunk 1 above)
Comment (d-bernat) [reply dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/generator/scenarios/aggregated.json]: done

File: dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/generator/scenarios/aggregated.json (hunk 2 above)
Comment (d-bernat) [reply dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/generator/scenarios/aggregated.json]: done

**Review by gnespolino — APPROVED** [review]

**Pull request closed without merging**